### PR TITLE
libgproj: Use SRID for PJ object creation when possible

### PR DESCRIPTION
**Pull request by HuidaeCho**
This PR tries to address #2257 by using SRID, not GRASS key-value pairs, for creating an PJ object. Here is [a related Q&A](https://geopandas.org/en/stable/docs/user_guide/projections.html#i-get-a-bound-crs).

### CODE REVIEW

File: lib/proj/get_proj.c [hunk 1]
```diff
@@ -249,7 +249,8 @@ int pj_get_kv(struct pj_info *info, const struct Key_Value *in_proj_keys,
     alloc_options("type=crs");
 #endif
     pjc = proj_context_create();
-    if (!(pj = proj_create_argv(pjc, nopt, opt_in))) {
+    if ((info->srid && !(pj = proj_create(pjc, info->srid))) ||
+        (!info->srid && !(pj = proj_create_argv(pjc, nopt, opt_in)))) {
```
Comment (metzm) [inline lib/proj/get_proj.c:253]: I recommend to first create a PJ object and then recycle the code at https://github.com/OSGeo/grass/blob/main/lib/proj/do_proj.c#L332 to check if it is a `PJ_TYPE_BOUND_CRS`, in which case the source crs should be retrieved.

Applies not only to `pj_get_kv()`, but also to  `pj_get_string()`

**Review by metzm — CHANGES_REQUESTED** [review]
The idea is correct, but this PR does not fix cases where no SRID is available and a bound CRS is provided. Should be easy to add.

**Comment by HuidaeCho** [comment]
@metzm Do we ever hit this PROJ error when `pj` is already not `NULL`?
https://github.com/OSGeo/grass/blob/830f31d215038c1a67112aa350619f652e04e70b/lib/proj/get_proj.c#L275-L278

**Comment by HuidaeCho** [comment]
> @metzm Do we ever hit this PROJ error when `pj` is already not `NULL`?
> 
> https://github.com/OSGeo/grass/blob/830f31d215038c1a67112aa350619f652e04e70b/lib/proj/get_proj.c#L275-L278

I think we can remove this check.

https://proj.org/development/reference/functions.html#c.proj_create_argv

> If creation of the transformation object fails, the function returns 0 and the PROJ error number is updated.

**Comment by metzm** [comment]
Apparently `proj_errno(pj)` can still return an error code, if `pj` is not NULL. At least this is used in various places in PROJ itself.

**Review by metzm — APPROVED** [review]
Thanks for the fix! This will not be the only one regarding the evolving PROJ lib I guess...

**Pull request merged**